### ISSUE TITLE
Is it wrong with  torch.arange()?

### ISSUE THREAD

**Issue by JuZiSYJ** [yinboc/liif#2]
Hi!
I try to test the **make_coord()** in utils.py

> def make_coord(shape, ranges=None, flatten=True):
    """ Make coordinates at grid centers.
    """
    coord_seqs = []
    for i, n in enumerate(shape):
        if ranges is None:
            v0, v1 = -1, 1
        else:
            v0, v1 = ranges[i]
        r = (v1 - v0) / (2 * n)
        seq = v0 + r + (2 * r) * **torch.arange(n)**
        coord_seqs.append(seq)
    ret = torch.stack(torch.meshgrid(*coord_seqs), dim=-1)
    if flatten:
        ret = ret.view(-1, ret.shape[-1])
    return ret

test is   0.5 * torch.arange(8),get

> tensor([0, 0, 0, 0, 0, 0, 0, 0])

what is wrong?

**Comment by yinboc** [comment]
Hi. This is not the expected behavior, what is your pytorch version?

**Comment by JuZiSYJ** [comment]
I test it in torch1.2, it is wrong
when I switch to torch1.3, the result is right

**Comment by yinboc** [comment]
That makes sense. As mentioned in the instructions, the code is tested under pytorch 1.6.0.
But thanks for reporting this! I will add a `.float()` to make it robust to pytorch versions.
